src/components/atoms/ConditionalWrapper/ConditionalWrapper.tsx

```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+
+export interface ConditionalWrapperProps {
+  condition: boolean;
+  wrapper: (children: React.ReactNode) => JSX.Element;
+}
+
+export const ConditionalWrapper = ({
+  condition,
+  wrapper,
+  children,
+}: React.PropsWithChildren<ConditionalWrapperProps>) => (condition ? wrapper(children) : <>{children}</>);
```

Comment:
```tsx
import { ComponentType, PropsWithChildren } from 'react';

export type ConditionalWrapperProps = {
  condition: boolean;
  Wrapper?: ComponentType<PropsWithChildren>;
} & PropsWithChildren;

export const ConditionalWrapper = ({
  condition,
  Wrapper = ({ children }) => children,
  children,
}: ConditionalWrapperProps) => (condition ? <Wrapper>{children}</Wrapper> : children);
```

Reply:
Good suggestion, causes some type errors though

<img width="1186" alt="image" src="https://github.com/ZopaPublic/react-components/assets/335607/9fe4f94c-09de-437a-bb81-a58040005143">

Also ZRC is not able to _not_ import React, we should update the `jsx` type in the `tsconfig.json` 👁️

Reply:
Updated :)